apps/backoffice-v2/src/common/hooks/useCurrentFilter/useCurrentFilter.ts

```diff
@@ -0,0 +1,16 @@
+import { useFilterId } from '@/common/hooks/useFilterId/useFilterId';
+import { useFiltersQuery } from '@/domains/filters/hooks/queries/useFiltersQuery/useFiltersQuery';
+import { useMemo } from 'react';
+
+export const useCurrentFilter = () => {
+  const filterId = useFilterId();
+  const { data: filters } = useFiltersQuery();
+
+  const currentFilter = useMemo(() => {
+    if (!Array.isArray(filters) || !filterId) return null;
```

Comment:
Please use `undefined` over `null` unless there's a use of the `null` value for the use-case.

Reply:
`undefined` - Is a compiler value and means that something isnt exists. Why we should use it if variable is declared and in use?

Reply:
In this context `currentFilter` doesn't exist. `null` introduces extra code, and mental overhead, and makes data more awkward to work with. We should ideally stick to either `undefined` or `null`, not both, `undefined` makes more sense since it is the returned value from operations like `find` and is already in use throughout the system.

Reply:
In current context value of `currentFilter` not exists but variable is declared and being processed by code. Also `null` return indicates that this value was returned by human and not compilier.

Methods such as `find` return `undefined` because returning `null` doesnt make sense.
`[].find(val => val === null)` // will return null in both cases no matter if value found or no.

If this is really concerning you then feel free to create poll & add such eslint rules

Reply:
Revisit PR post poll.